extensions/websockets-next/deployment/src/test/java/io/quarkus/websockets/next/test/security/HttpUpgradeAuthenticatedAnnotationTest.java

```diff
@@ -0,0 +1,147 @@
+                            PublicEndpoint.class, PublicEndpoint.SubEndpoint.class));
+
+    @TestHTTPResource("public-end")
+    URI publicEndUri;
+
+    @TestHTTPResource("public-end/sub")
+    URI subEndUri;
+
+    @Test
+    public void testSubEndpoint() {
+        try (WSClient client = new WSClient(vertx)) {
+            client.connect(publicEndUri);
+            client.waitForMessages(1);
+            assertEquals("ready", client.getMessages().get(0).toString());
+            client.sendAndAwait("hello");
+            client.waitForMessages(2);
+            assertEquals("hello", client.getMessages().get(1).toString());
+        }
+
+        try (WSClient client = new WSClient(vertx)) {
+            CompletionException ce = assertThrows(CompletionException.class, () -> client.connect(subEndUri));
+            Throwable root = ExceptionUtil.getRootCause(ce);
+            assertInstanceOf(UpgradeRejectedException.class, root);
+            assertTrue(root.getMessage().contains("401"), root.getMessage());
+        }
+        try (WSClient client = new WSClient(vertx)) {
+            client.connect(basicAuth("admin", "admin"), subEndUri);
+            client.waitForMessages(1);
+            assertEquals("ready", client.getMessages().get(0).toString());
+            client.sendAndAwait("hello");
+            client.waitForMessages(2);
+            assertEquals("sub-endpoint", client.getMessages().get(1).toString());
+        }
+        try (WSClient client = new WSClient(vertx)) {
+            client.connect(basicAuth("user", "user"), subEndUri);
+            client.waitForMessages(1);
+            assertEquals("ready", client.getMessages().get(0).toString());
+            client.sendAndAwait("hello");
+            client.waitForMessages(2);
+            assertEquals("sub-endpoint:forbidden:user", client.getMessages().get(1).toString());
+        }
+    }
+
+    @WebSocket(path = "/public-end")
+    public static class PublicEndpoint {
+
+        @OnOpen
+        String open() {
+            return "ready";
+        }
+
+        @OnTextMessage
+        String echo(String message) {
+            return message;
+        }
+
+        @Authenticated
+        @WebSocket(path = "/sub")
+        public static class SubEndpoint {
+
+            @Inject
+            CurrentIdentityAssociation currentIdentity;
+
+            @OnOpen
+            String open() {
+                return "ready";
+            }
+
+            @RolesAllowed("admin")
+            @OnTextMessage
+            String echo(String message) {
+                return "sub-endpoint";
+            }
+
+            @OnError
+            String error(ForbiddenException t) {
+                return "sub-endpoint:forbidden:" + currentIdentity.getIdentity().getPrincipal().getName();
+            }
+        }
+
+    }
+
+    @Authenticated
+    @WebSocket(path = "/end")
+    public static class Endpoint {
```

Comment:
Unused?

Reply:
yes, it is tested by SecurityTestBase and unit tests declared directly in this test class are just complementing the base tests

Reply:
please see `extends SecurityTestBase`